### PR TITLE
Feature/dhcp pool algorithm

### PR DESCRIPTION
**Pull request by fdurand**
# Description
Add algorithm of the dhcp pool.
- Random algorithm (default one)
- Give the oldest given ip address

# Impacts
DHCP Server

# Delete branch after merge
YES

# Checklist
- [n/a ] Document the feature
- [n/a ] Add unit tests
- [n/a ] Add acceptance tests (TestLink)

# NEWS file entries
## New Features
* Added dhcp pool algorithm (Random/Oldest ip)

### CODE REVIEW

File: go/dhcp/pool/pool.go [hunk 1]
```diff
@@ -105,14 +113,38 @@ func (dp *DHCPPool) GetFreeIPIndex(mac string) (uint64, string, error) {
 	if len(dp.free) == 0 {
 		return 0, FreeMac, errors.New("DHCP pool is full")
 	}
-	index := rand.Intn(len(dp.free))
 
 	var available uint64
-	for available = range dp.free {
-		if index == 0 {
+
+	if dp.algorithm == 2 {
```
Comment (julsemaan) [inline go/dhcp/pool/pool.go]: Use a constant

File: go/dhcp/pool/pool_test.go [hunk 2]
```diff
@@ -6,7 +6,8 @@ import (
 
 func TestReserveIPIndex(t *testing.T) {
 	cap := uint64(5)
-	dp := NewDHCPPool(cap)
+	algo := int(1)
```
Comment (julsemaan) [inline go/dhcp/pool/pool_test.go]: Use a constant for this and the ones below

**Comment by fdurand** [comment]
Bump

**Comment by julsemaan** [comment]
I think the algorithm field is missing from the new admin

**Comment by fdurand** [comment]
Bump

**Pull request merged**